### DISCUSSION TITLE
Problem with converting large images to .AVIF

### DISCUSSION THREAD

**Discussion by FrontierDK** [ImageMagick/ImageMagick#4944] [Development]
Hi all.

When I use the x86 version of the latest Imagemagick, it won't make a working file - if the original jpeg is rather large. This is a sample image from a Samsung Galaxy S20 phone: http://mitip.dk/2020-02-20-133941_49563323986_o.jpg

Command used: magick.exe 2020-02-20-133941_49563323986_o.jpg new.avif

**Comment by snibgo** [comment]
Please state the actual version of IM (the output from "magick -version") and your platform.

With pre-built IM 7.1.0-4 on Windows 8.1, I can create AVIF files successfully.

With the linked JPEG ...
```
magick -verbose 2020-02-20-133941_49563323986_o.jpg x.avif
```
... this eats all available memory, creating no temporary files, and finishes. When I try to read the created avif, I get:
```
f:\web\im>%IMG7%magick x.avif info:

magick: Invalid input: Item has no data: Item with ID 1 has no compressed data `
x.avif' @ error/heic.c/IsHeifSuccess/135.
```
The same problem occurs when my input is "-size 12000x9000 xc:", so the problem isn't the JPEG.

Perhaps the first command ran out of memory, but IM did not realize that, and a corrupt file was created.

**Reply by FrontierDK** [reply]
Hi there, thank you for your response.

Here is my version info:
Version: ImageMagick 7.1.0-27 Q16-HDRI x86 2022-03-05 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193131104)

I can also convert smaller files to .avif without any issues. But...this large file can't be converted to .avif. When I try to convert, here is what I get:
magick -verbose d:\2020-02-20-133941_49563323986_o.jpg d:\2020-02-20-133941_49563323986_o.avif
d:\2020-02-20-133941_49563323986_o.jpg JPEG 12000x9000 12000x9000+0+0 8-bit sRGB 31.3527MiB 2.906u 0:02.901
d:\2020-02-20-133941_49563323986_o.jpg=>d:\2020-02-20-133941_49563323986_o.avif JPEG 12000x9000 12000x9000+0+0 8-bit sRGB 130492B 13.281u 0:16.503

The result is a small file which isn't a valid .avif image.

**Comment by urban-warrior** [comment]
Try this command (works for us):

```
magick -limit memory 0 -limit map 0 2020-02-20-133941_49563323986_o.jpg x.avif
```

ImageMagick utilizes the libheif delegate library to read/write the AVIF format.  It appears to want to allocate the entire image in memory, whereas ImageMagick can cache the image on disk to reduce the memory footprint.

**Comment by snibgo** [comment]
```
f:\web\im>%IMG7%magick -verbose -limit memory 0 -limit map 0 -size 12000x9000 xc: x2.avif
xc:=> XC 12000x9000 12000x9000+0+0 16-bit sRGB 2.141u 0:02.788
xc:=>x2.avif XC 12000x9000 12000x9000+0+0 16-bit sRGB 252B 38.688u 0:54.825

f:\web\im>%IMG7%magick x2.avif info:
magick: Invalid input: Item has no data: Item with ID 1 has no compressed data `
x2.avif' @ error/heic.c/IsHeifSuccess/135.
```
The first command creates a temporary file of 1,296,000,000 bytes (ie 12000x9000 pixels at 12 bytes/pixel), so that is good. But it still eats all available memory of my 12 GB laptop, and the output AVIF is not valid.

EDIT to add:
```
f:\web\im>%IMG7%magick -list format |grep -i avif

     AVIF* HEIC      rw+   AV1 Image File Format (1.12.0)
```

**Reply by urban-warrior** [reply]
We're using libheif, version 1.12.0.  Your command works fine on our Fedora Linux host with 32GB of memory.  Try limits on your `info:` command:

```
magick -verbose -limit memory 0 -limit map 0 x2.avif info:
```

Does that work for you?

**Comment by snibgo** [comment]
Sadly, no.
```
f:\web\im>%IMG7%magick -verbose -limit memory 0 -limit map 0 x2.avif info:

magick: Invalid input: Item has no data: Item with ID 1 has no compressed data `
x2.avif' @ error/heic.c/IsHeifSuccess/135.
```

EDIT: I have uploaded x2.avif to  http://im.snibgo.com/x2.avif

It has 252 bytes.

**Comment by urban-warrior** [comment]
@dlemstra, can you take a look @ libheif under Windows.  Is this a Windows build issue?

**Comment by dlemstra** [comment]
Are you using the x86 or the x64 version of InageMagick @snibgo?

**Comment by snibgo** [comment]
x64.
```
f:\web\im>%IMG7%magick -version
Version: ImageMagick 7.1.0-4 Q16 x64 2021-07-18 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192930038
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms
 lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
```
This is a rather old version.

**Comment by dlemstra** [comment]
The `http://mitip.dk/2020-02-20-133941_49563323986_o.jpg` file is no longer available. Can someone attach that file to this question?

**Reply by FrontierDK** [reply]
Sorry, thought this topic wasn't going to be addressed, so I removed the file. It's back now.

**Comment by dlemstra** [comment]
Archiving the file (added bogus `.zip` extension):

[2020-02-20-133941_49563323986_o.7z.001.zip](https://github.com/ImageMagick/ImageMagick/files/8620911/2020-02-20-133941_49563323986_o.7z.001.zip)
[2020-02-20-133941_49563323986_o.7z.002.zip](https://github.com/ImageMagick/ImageMagick/files/8620915/2020-02-20-133941_49563323986_o.7z.002.zip)
[2020-02-20-133941_49563323986_o.7z.003.zip](https://github.com/ImageMagick/ImageMagick/files/8620912/2020-02-20-133941_49563323986_o.7z.003.zip)
[2020-02-20-133941_49563323986_o.7z.004.zip](https://github.com/ImageMagick/ImageMagick/files/8620914/2020-02-20-133941_49563323986_o.7z.004.zip)
